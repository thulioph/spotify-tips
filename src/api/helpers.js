import axios from 'axios';

export const requestPromise = (url) => {
  if (!url) return false;

  return new Promise((resolve, reject) => {
    axios.get(url).then(success => resolve(success)).catch(err => reject(err));
  });
};