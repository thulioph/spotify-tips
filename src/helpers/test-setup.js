import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// ====

const exposedProperties = ['window', 'navigator', 'document'];

global.window = document.defaultView;

Object.keys(document.defaultView).forEach((property) => {
    if (typeof global[property] === 'undefined') {
        exposedProperties.push(property);
        global[property] = document.defaultView[property];
    }
});

global.navigator = {
    userAgent: 'node.js'
};

// ====

Enzyme.configure({ adapter: new Adapter() });