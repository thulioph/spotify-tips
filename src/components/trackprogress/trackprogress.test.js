import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import chai, { expect as cExpect } from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import '../../helpers/test-setup';
import TrackProgress from './index';

// ====

chai.use(sinonChai);

// ====

describe('<TrackProgress />', () => {
    describe('Snapshot', () => {
        it('should render without crashing.', () => {
            const wrapper = shallow(<TrackProgress />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('Structure', () => {
        it('should have a div when mount', () => {
            const wrapper = shallow(<TrackProgress />);
            cExpect(wrapper.find('div')).to.have.length(1);
        });
    });

    describe('Props', () => {
        it('should render with the passed time.', () => {
            const wrapper = shallow(
                <TrackProgress time={'50%'} />
            );

            cExpect(wrapper.find('div').props().style.width).to.be.eql('50%');
            
            const wrapper2 = shallow(
                <TrackProgress time={'92%'} />
            );

            cExpect(wrapper2.find('div').props().style.width).to.be.eql('92%');
        });
    });
});