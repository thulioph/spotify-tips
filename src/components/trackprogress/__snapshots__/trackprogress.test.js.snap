// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackProgress /> Snapshot should render without crashing. 1`] = `
<div
  className="loading-bar"
  style={
    Object {
      "width": undefined,
    }
  }
/>
`;
